### PR TITLE
Added code to auto-detect webroot and broke stuff while I was at it

### PR DESCRIPTION
**Pull request by linedash**

### CODE REVIEW

**Comment by linedash** [comment]
The syntax for the sys.argv stuff is now likely wrong and causing breakage.
-
main should be moved below findlwngwebroot().   
-
findlwngwebroot using messy code; but I can actually understand what it's doing.
-
Think am calling findlwngwebroot in the wrong place, as it spits out multiple itterations of 'lwng' per run.
-
webroots variable reference now causes issues:

Traceback (most recent call last):
  File "./logfilter.py", line 158, in <module>
    sys.exit(main())
  File "./logfilter.py", line 127, in main
    for item in filter_logs(follow_logs(glob_logs(webroots)), geoip_db):
UnboundLocalError: local variable 'webroots' referenced before assignment

File: logfilter.py [hunk 1]
```diff
@@ -107,12 +108,19 @@ def geoip_parse(line):
 
 
 def main():
-    if len(sys.argv) < 2:
-        print >> sys.stderr, "Please provide a webroots path."
-        return 1
+    #auto-detect host type and set webroot accordingly.  Also accept from first argument
+    if findhost.main() == "linweb":
```
Comment (kgaughan) [inline logfilter.py]: Change the code to this:

    host_type = findhost.get_host_type(socket.getfqdn())
    if host_type == 'linweb':
        # ...
    elif host_type == 'lwng':
        # ...
    else:
        # ...

File: logfilter.py [hunk 2]
```diff
@@ -122,5 +130,29 @@ def main():
     return 0
 
 
+
+r'''
+I don't care if this is done wrong.  Leave it alone.
+'''
+
+def findlwngwebroot():
+    p1 = subprocess.Popen(["redis-cli", '-s', '/var/lib/redis/redis.sock', 'KEYS', 'website-by-name:*'], stdout=subprocess.PIPE)
```
Comment (kgaughan) [inline logfilter.py]: Check what I'm doing here: https://gist.github.com/kgaughan/894d87bfb46d5ade76a7432f7a57e018

If you want to get the first line  of the response of some command back, the code in `redis_query_value` is doing the right thing.

**Comment by linedash** [comment]
Fixed code chunks referenced

File: logfilter.py [hunk 3]
```diff
@@ -107,20 +108,52 @@ def geoip_parse(line):
...
+
+    #if len(sys.argv) < 2:
+    #    print >> sys.stderr, "Please provide a webroots path."
+    #    return 1
 
-    # Get the webroots path as first argument
-    webroots = sys.argv[1]
     # Get the GeoIP database path as the second argument, if present.
-    geoip_db = 'GeoIP.dat' if len(sys.argv) == 2 else sys.argv[2]
+    geoip_db = sys.argv[2] if len(sys.argv) >= 2 else "GeoIP.dat"
```
Comment (kgaughan) [inline logfilter.py]: Argh! Change the `>= 2` to `> 2`. My bad.

**Pull request merged**